sdk/tests/conformance/glsl/misc/shader-uniform-packing-restrictions.html

```diff
@@ -262,6 +262,78 @@
       linkSuccess: true,
       passMsg: shaderType.type + " shader with " + minVars + " uniforms of " + info.type + " (the minimum required) should succeed",
     });
+    
+    if (maxPerRow > 1) {
+      
+      var generateCodeArray = function() {
+        var uniforms = [];
+        var sumTerms = [];
+        var itemcount = Math.max(Math.floor(shaderType.maxVectors / maxPerRow) + 1, minVars ); 
+        for (var uu = 0; uu < maxPerRow; ++uu) {
+          uniforms.push("    uniform " + info.type + " u_uniform" + uu + "[" + itemcount + "];");
+          sumTerms.push(wtu.replaceParams(info.uniToF, {id: uu, index: "[" + (itemcount - 1) + "]"}));
+        }
+        return {
+          uniforms: uniforms.join("\n"),
+          code: info.fType + " sum = " + sumTerms.join(" + \n            ") + ";",
+          result: wtu.replaceParams(info.fToVec4, {f: 'sum'})
+        };
+      };
+      
+      // Test required uniforms of type.
+      tests.push({
+        vShaderSource: wtu.replaceParams(shaderType.vertUniformTest, generateCodeArray(), info),
+        vShaderSuccess: true,
+        fShaderSource: wtu.replaceParams(shaderType.fragUniformTest, generateCodeArray(), info),
+        fShaderSuccess: true,
+        linkSuccess: true,
+        passMsg: shaderType.type + " shader with " + maxPerRow + "uniform arrays of " + info.type + " with " + (Math.floor(shaderType.maxVectors / maxPerRow) + 1) + " elements should succeed",
```

Comment:
The Math.floor expression here looks like it's supposed to be identical to "itemcount" but the code's duplicated. Please share it.